internal/command/deploy/machines.go

```diff
@@ -539,18 +617,34 @@ func (md *machineDeployment) resolveUpdatedMachineConfig(origMachineRaw *api.Mac
 			Config: &api.MachineConfig{},
 		}
 	}
+	newConfig := machine.CloneConfig(origMachineRaw.Config)
 
 	launchInput := &api.LaunchMachineInput{
 		ID:      origMachineRaw.ID,
 		AppID:   md.app.Name,
 		OrgSlug: md.app.Organization.ID,
-		Config:  lo.Ternary(md.restartOnly, origMachineRaw.Config, &api.MachineConfig{}),
+		Config:  newConfig,
 		Region:  origMachineRaw.Region,
 	}
 
-	launchInput.Config.Metadata = md.defaultMachineMetadata()
-	for k, v := range origMachineRaw.Config.Metadata {
-		if !isFlyAppsPlatformMetadata(k) {
+	launchInput.Config.Statics = nil
```

Comment:
it's suppose to be pedantic comment but this is a bug, because it removes statics when it is is in "restart only" mode. I'd move it right above the block that process statics:

```
for _, s := range md.appConfig.Statics {
		launchInput.Config.Statics = append(launchInput.Config.Statics, &api.Static{
			GuestPath: s.GuestPath,
			UrlPrefix: s.UrlPrefix,
		})
	}
```

Reply:
I had it moved up so that it would clear for release commands too, I forgot about restartOnly. There's quite a few distinct states to keep track of here...

Reply:
indeed, it is asking for a redo but without tests it is too risky to refactor this logic  now